### PR TITLE
Fix Mandatory fields are not marked when reCaptcha is enabled

### PR DESCRIPTION
**Pull request by hwupathum**
fixes https://github.com/wso2/product-is/issues/14608
fixes https://github.com/wso2/product-is/issues/14563
### Purpose

-  Made firstname field required in username-recovery flow
image
- Fixed the issue where the validations didn't work in self-resigration, username recovery and password recovery flows

### CODE REVIEW

File: apps/recovery-portal/src/main/webapp/challenge-question-view.jsp [hunk 1]
```diff
@@ -153,9 +160,23 @@
...
-            function onSubmit(token) {
-               $("#securityQuestionForm").submit();
+            $("#securityQuestionForm").submit(function (e) {
+               <%
+                   if (reCaptchaEnabled) {
+               %>
+               if (!grecaptcha.getResponse()) {
+                   e.preventDefault();
+                   grecaptcha.execute();
+                   return true;
```
Comment (thivi) [inline apps/recovery-portal/src/main/webapp/challenge-question-view.jsp:174]:
```suggestion

                   return true;
```

File: apps/recovery-portal/src/main/webapp/challenge-questions-view-all.jsp [hunk 2]
```diff
@@ -162,9 +169,23 @@
...
-        function onSubmit(token) {
-           $("#securityQuestionForm").submit();
+        $("#securityQuestionForm").submit(function (e) {
+           <%
+               if (reCaptchaEnabled) {
+           %>
+           if (!grecaptcha.getResponse()) {
+               e.preventDefault();
+               grecaptcha.execute();
+               return true;
```
Comment (thivi) [inline apps/recovery-portal/src/main/webapp/challenge-questions-view-all.jsp:183]:
```suggestion

               return true;
```

File: apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp [hunk 3]
```diff
@@ -331,13 +336,26 @@
...
 
             $("#recoverDetailsForm").submit(function (e) {
+
+                <%
+                    if (reCaptchaEnabled) {
+                %>
+                if (!grecaptcha.getResponse()) {
+                    e.preventDefault();
+                    grecaptcha.execute();
+                    return true;
```
Comment (thivi) [inline apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp:354]:
```suggestion

                    return true;
```

File: apps/recovery-portal/src/main/webapp/password-recovery.jsp [hunk 4]
```diff
@@ -328,14 +333,26 @@
...
 
             $("#recoverDetailsForm").submit(function (e) {
 
+                <%
+                    if (reCaptchaEnabled) {
+                %>
+                if (!grecaptcha.getResponse()) {
+                    e.preventDefault();
+                    grecaptcha.execute();
+                    return true;
```
Comment (thivi) [inline apps/recovery-portal/src/main/webapp/password-recovery.jsp:350]:
```suggestion

                    return true;
```

File: apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp [hunk 5]
```diff
@@ -800,6 +805,18 @@
...
 
             $("#register").submit(function (e) {
+
+                <%
+                    if (reCaptchaEnabled) {
+                %>
+                if (!grecaptcha.getResponse()) {
+                    e.preventDefault();
+                    grecaptcha.execute();
+                    return true;
```
Comment (thivi) [inline apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp:815]:
```suggestion

                    return true;
```

File: apps/recovery-portal/src/main/webapp/self-registration-without-verification.jsp [hunk 6]
```diff
@@ -285,10 +290,26 @@
...
 
             $("#register").submit(function (e) {
 
+                <%
+                    if (reCaptchaEnabled) {
+                %>
+                if (!grecaptcha.getResponse()) {
+                    e.preventDefault();
+                    grecaptcha.execute();
+                    return true;
```
Comment (thivi) [inline apps/recovery-portal/src/main/webapp/self-registration-without-verification.jsp:307]:
```suggestion

                    return true;
```

File: apps/recovery-portal/src/main/webapp/username-recovery.jsp [hunk 7]
```diff
@@ -321,13 +325,24 @@
...
         $(document).ready(function () {
 
             $("#recoverDetailsForm").submit(function (e) {
+                <%
+                    if (reCaptchaEnabled) {
+                %>
+                if (!grecaptcha.getResponse()) {
+                    e.preventDefault();
+                    grecaptcha.execute();
+                    return true;
```
Comment (thivi) [inline apps/recovery-portal/src/main/webapp/username-recovery.jsp:341]:
```suggestion

                    return true;
```

**Review by thivi — APPROVED** [review]

**Review by piraveena — APPROVED** [review]

File: apps/recovery-portal/src/main/webapp/challenge-question-view.jsp [hunk 8]
```diff
@@ -109,19 +109,26 @@
...
+                            if (reCaptchaEnabled) {
+                                String reCaptchaKey = CaptchaUtil.reCaptchaSiteKey();
+                        %>
+                        <div class="field">
+                            <div class="g-recaptcha"
+                                    data-size="invisible"
+                                    data-callback="onCompleted"
+                                    data-action="usernameRecovery"
+                                    data-sitekey=
+                                            "<%=Encode.forHtmlContent(reCaptchaKey)%>">
```
Comment (chamathns) [inline apps/recovery-portal/src/main/webapp/challenge-question-view.jsp:122]: check formatting here

File: apps/recovery-portal/src/main/webapp/challenge-questions-view-all.jsp [hunk 9]
```diff
@@ -122,19 +122,26 @@
...
+                            if (reCaptchaEnabled) {
+                                String reCaptchaKey = CaptchaUtil.reCaptchaSiteKey();
+                        %>
+                        <div class="field">
+                            <div class="g-recaptcha"
+                                    data-size="invisible"
+                                    data-callback="onCompleted"
+                                    data-action="securityQuestion"
+                                    data-sitekey=
+                                            "<%=Encode.forHtmlContent(reCaptchaKey)%>">
```
Comment (chamathns) [inline apps/recovery-portal/src/main/webapp/challenge-questions-view-all.jsp:135]: check formatting

Comment (chamathns) [reply apps/recovery-portal/src/main/webapp/challenge-questions-view-all.jsp:135]: Fix other places as well if applicable

File: apps/recovery-portal/src/main/webapp/challenge-question-view.jsp (hunk 8 above)
Comment (hwupathum) [reply apps/recovery-portal/src/main/webapp/challenge-question-view.jsp:122]: Addressed by https://github.com/wso2/identity-apps/pull/3817

File: apps/recovery-portal/src/main/webapp/challenge-questions-view-all.jsp (hunk 9 above)
Comment (hwupathum) [reply apps/recovery-portal/src/main/webapp/challenge-questions-view-all.jsp:135]: Addressed by https://github.com/wso2/identity-apps/pull/3817

**Pull request merged**